internal/converter/internal/promtailconvert/promtailconvert_test.go

```diff
@@ -6,8 +6,88 @@ import (
 	"github.com/grafana/agent/internal/converter/internal/promtailconvert"
 	"github.com/grafana/agent/internal/converter/internal/test_common"
 	_ "github.com/grafana/agent/internal/static/metrics/instance" // Imported to override default values via the init function.
+	"github.com/stretchr/testify/require"
 )
 
 func TestConvert(t *testing.T) {
 	test_common.TestDirectory(t, "testdata", ".yaml", true, []string{}, promtailconvert.Convert)
 }
+
+// TestGlobalPositions will ensure that legacy is set if there are multiple readers. Since the global positions file contains
+// all files but when converted only the files of the actual individual jobs should be tracked. This does lead to a case
+// where there are orphaned files but since they will not be read by the job they are noops. If they add the previously unused
+// filepath to a job, say for instance add the fun paths to fun2 then it will pick up where it left off from the conversion
+// but this is likely the best path and unlikely to be a common occurrence.
+func TestGlobalPositions(t *testing.T) {
```

Comment:
can we do this using the existing framework for testing with a new river/yaml combo in the testdata directory?